### ISSUE TITLE
Add a basic Music Player interface and use either spotify free sdk or some other free api to play songs listed on the app

### ISSUE THREAD

**Issue by GeekyPS** [GeekyPS/SpotiStats_flutter#4]
The music player UI need not be too fancy just a simple basic interface would do, but the music pause play functionality should be working.

PS: This is a bonus issue carrying 45 points

**Comment by chandansgowda** [comment]
Hey @GeekyPS 
Shall I take this up?

**Comment by GeekyPS** [comment]
Yeah sure, you can contact me anytime you need help or want resources . Best of luck on the issue. 
Happy Contributing! :fire:

**Comment by chandansgowda** [comment]
Hey @GeekyPS 
Thanks for assigning me.
I would need the following from you:

**FOR Android:**
- Register your app in the [spotify developer portal](https://developer.spotify.com/dashboard/). You also need to create a sha-1 fingerprint and add this and your package name to the app settings on the dashboard as well as a redirect url.
- Follow the instructions in the section Setup the iOS SDK of [Spotify Android SDK Quick Start](https://developer.spotify.com/documentation/android/quick-start/).

**FOR iOS:**
- Register your app in the [spotify developer portal](https://developer.spotify.com/dashboard/). You also need to register your Bundle ID as well as a Redirect URI.
- Follow the instructions in the section Setup the iOS SDK of [Spotify iOS SDK Quick Start](https://developer.spotify.com/documentation/ios/quick-start/).

**Comment by GeekyPS** [comment]
Surely, I can do those things but for your implementation use your Spotify developer account and app to test the code, when  making a pull request delete all the confidential credentials and write a clear description of the code setup

The Spotify Android and IOS SDK are for native languages Kotlin and Swift C, if you know how to execute native code in flutter go on , otherwise you can take help of this package https://pub.dev/packages/spotify_sdk on pub.dev

**Comment by chandansgowda** [comment]
Yeah. Sure! I'll be using spotify_sdk package.

**Comment by chandansgowda** [comment]
Hey @GeekyPS ,
I worked on this feature and found out that the project has to be migrated to mimimum Dart SDK version 2.12.0 but right now, it's 2.11.0 . Shall I migrate the project to 2.12.0 min SDK ?? If yes please create an issue and assign me for the same. Also, to play a song I will need the spotify uri of the same. Please modify the backend accordingly.

Thanks

**Comment by GeekyPS** [comment]
Yeah sure, i am creating issue to upgrade all libraries and deleting redundant code , but this would be a easy issue hence carry 5 points , also since you are using spotify sdk so ig you won't be needing backend but if you do can u explain in detail about what you need

**Comment by GeekyPS** [comment]
Created the issue , comment on it in order to get assigned

**Comment by chandansgowda** [comment]
Done
